src/core/pointcloud/qgspointcloudattribute.cpp

```diff
@@ -69,18 +54,17 @@ void QgsPointCloudAttribute::updateSize()
 QgsPointCloudAttributeCollection::QgsPointCloudAttributeCollection() = default;
 
 QgsPointCloudAttributeCollection::QgsPointCloudAttributeCollection( const QVector<QgsPointCloudAttribute> &attributes )
-  : mAttributes( attributes )
 {
-  for ( int i = 0; i < mAttributes.size(); ++i )
+  mAttributes.reserve( attributes.size() );
+  for ( const QgsPointCloudAttribute &attribute : attributes )
   {
-    const QgsPointCloudAttribute &attribute = mAttributes.at( i );
-    mSize += attribute.size();
+    push_back( attribute );
   }
```

Comment:
you need to initialize mCachedAttributes and mSize in this constructor too

Reply:
That should be handled inside push_back, right?

Reply:
of course, misread it 🤦